symfony/plugins/orangehrmAttendancePlugin/Api/Model/DetailedAttendanceRecordModel.php

```diff
@@ -0,0 +1,96 @@
+ *
+ * OrangeHRM is free software; you can redistribute it and/or modify it under the terms of
+ * the GNU General Public License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * OrangeHRM is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+
+namespace OrangeHRM\Attendance\Api\Model;
+
+use OrangeHRM\Core\Api\V2\Serializer\ModelTrait;
+use OrangeHRM\Core\Api\V2\Serializer\Normalizable;
+use OrangeHRM\Entity\AttendanceRecord;
+
+class DetailedAttendanceRecordModel implements Normalizable
+{
+    use ModelTrait;
+
+    public function __construct(AttendanceRecord $attendanceRecord)
+    {
+        $this->setEntity($attendanceRecord);
+        $this->setFilters(
+            [
+                'id',
+                ['getDecorator', 'getPunchInUTCDate'],
+                ['getDecorator', 'getPunchInUTCTime'],
+                ['getDecorator', 'getPunchInUserDate'],
+                ['getDecorator', 'getPunchInUserTime'],
+                'punchInTimeOffset',
+                'punchInTimezoneName',
+                ['getDecorator', 'getPunchInTimezoneLabel'],
+                'punchInTimeOffset',
+                'punchInNote',
+                ['getDecorator', 'getPunchOutUTCDate'],
+                ['getDecorator', 'getPunchOutUTCTime'],
+                ['getDecorator', 'getPunchOutUserDate'],
+                ['getDecorator', 'getPunchOutUserTime'],
+                'punchOutTimeOffset',
+                'punchOutTimezoneName',
+                ['getDecorator', 'getPunchOutTimezoneLabel'],
+                'punchOutTimeOffset',
+                'punchOutNote',
+                'state',
+                ['getDecorator', 'getAttendanceState'],
+                ['getEmployee', 'getEmpNumber'],
+                ['getEmployee', 'getLastName'],
+                ['getEmployee', 'getFirstName'],
+                ['getEmployee', 'getMiddleName'],
+                ['getEmployee', 'getEmployeeId'],
+                ['getEmployee', 'getEmployeeTerminationRecord', 'getId']
+            ]
+        );
+        $this->setAttributeNames(
+            [
+                'id',
+                ['punchIn', 'utcDate'],
+                ['punchIn', 'utcTime'],
+                ['punchIn', 'userDate'],
+                ['punchIn', 'userTime'],
+                ['punchIn', 'timezoneOffset'],
+                ['punchIn', 'timezone','name'],
+                ['punchIn', 'timezone','label'],
+                ['punchIn', 'timezone','_offset'],
```

Comment:
not clear

Reply:
this is to directly set timezone autocomplete once we fetch result by specific record id. we have to set autocomplete directly to v-model from attendanceRecord.punchIn.timezone, attendanceRecord.punchOut.timezone.this was also discussed with chamara